service/src/main/java/bio/terra/workspace/service/resource/controlled/cloud/azure/storageContainer/DeleteAzureStorageContainerStep.java

```diff
@@ -48,24 +60,63 @@ public StepResult doStep(FlightContext context) throws InterruptedException {
     final StorageManager manager = crlService.getStorageManager(azureCloudContext, azureConfig);
 
     try {
-      WsmResource wsmResource =
-          resourceDao.getResource(resource.getWorkspaceId(), resource.getStorageAccountId());
-      ControlledAzureStorageResource storageAccount =
-          wsmResource
-              .castToControlledResource()
-              .castByEnum(WsmResourceType.CONTROLLED_AZURE_STORAGE_ACCOUNT);
+      String storageAccountName;
+      if (resource.getStorageAccountId() != null) {
```

Comment:
I think it would be simpler to store the storage account name with the container during the create, rather than having to do all of this to locate it on delete. The name must be immutable, right?

Reply:
That is a good suggestion! But it looks like it will require changes in core logic for CreateControlledResourceFlight: metadata store step is very first in the chain of steps. At that moment we don't have information about storage account name yet. We calculate it later in Verify step. Maybe we can consider additional step which updates metadata. I think it's probably good to implement as a separate PR if we decide to do so.